### ISSUE TITLE
Stack Image Id Synchronization example

### ISSUE THREAD

**Issue by rjsgml5698** [cornerstonejs/cornerstoneTools#1184]
Can I get a hint about the '**Stack Image Id Synchronization**' example that is still disabled in the next link?

I'm trying using the library '**react-cornerstone-viewport**'

Link :  https://tools.cornerstonejs.org/examples/

thank you!

**Comment by dannyrb** [comment]
A synchronizer:

- Is tied to a specific event
- Has a handler for that event
- Has "source" enabled elements
- Has "target" enabled elements

When the event is emitted from a "source" element, the handler is used. Often updating/synchronizing the "target" elements.

Hopefully that demystifies them a bit. You can create a similar effect in react-cornerstone-viewport by using the eventListeners prop. Hook each viewport into the same handler (probably cornerstone's new image event), and use the event data to update the other viewports.

The handler for image ID synch is hiding in the repos source.

**Comment by rjsgml5698** [comment]
Thanks for the explanation!!

Could you explain it in more detail? Do you have any simple examples?

I still don't understand it enough to implement it in code

I really need your help

thank you.

**Comment by kuehlc** [comment]
I would also appreciate a simple example if possible. I don't know about the cornerstone-viewport thing, but I just need to be able to activate stack scroll sync for any number of dicoms if possible.

**Comment by dannyrb** [comment]
> How do I find the 'sourceElement' and 'targetElement'?

Use the `onElementEnabled` hook to grab a reference to the EnabledElement. It's returned by the callback for `react-cornerstone-viewport`

> I just need to be able to activate stack scroll sync for any number of dicoms if possible.

What are you trying to sync? A synchronizer is really just an event handler that tracks "target elements", "source element", and has an on/off switch. So to synchronize something, listen for an event, and then based on what changed, update all (or a subset) of the other viewports.

**Comment by kuehlc** [comment]
@dannyrb I am trying to make it so that when I scroll through one of the dicoms stacks, all other dicoms also scroll at the same time. As a simple example with your other tools, I want to do what is shown on this page that has wwwc sync except with stack scrolling instead: https://tools.cornerstonejs.org/examples/tools/wwwc-sync.html

**Comment by dannyrb** [comment]
This event:
https://github.com/cornerstonejs/cornerstoneTools/blob/master/src/events.js#L199

`cornerstonetoolsstackscroll`  contains `direction`, which can be used with the modified element's current "stack state" to determine the new index and how much to increment/scroll the other viewports: 

https://github.com/cornerstonejs/cornerstoneTools/blob/3cd6d0c567a4915606225564e148db347c63c436/src/util/scrollToIndex.js#L95-L98

This handler:
https://github.com/cornerstonejs/cornerstoneTools/blob/master/src/synchronization/stackScrollSynchronizer.js

**Comment by kuehlc** [comment]
@dannyrb  `const synchronizer = new cornerstoneTools.Synchronizer(
      'cornerstoneimagerendered',
      cornerstoneTools.stackScrollSynchronizer
      // StackScrollTool
    );
    for (let i = 0; i < viewerArray.length; i++) {
      const element = document.getElementById('viewer' + i);
      dicomElements.push(element);
      synchronizer.add(element);
    }
    synchronizer.enabled = true;

    dicomElements.forEach(element => {
      // cornerstoneTools.addTool(tool);
      const StackScrollMouseWheelTool = cornerstoneTools.StackScrollMouseWheelTool;
      cornerstoneTools.addTool(StackScrollMouseWheelTool);
      cornerstoneTools.setToolActive('StackScrollMouseWheel', {
        mouseButtonMask: { },
        synchronizationContext: synchronizer,
      });
    });`
This is what I tried before, but obviously that didn't work. I'll try to use the direction thing that you mentioned. A simple example here or on the doc page would still be nice if you have time and it's  not to much trouble though. Thanks for the tips.

**Comment by dannyrb** [comment]
It's a bit odd. Most of the `cornerstone` work isn't funded by a grant, so it's usually updated as a side-effect of needs/enhancements on projects like OHIF. I try to volunteer some of my time, but I've been a bit stretched thin these days.

I've been working on: https://ohiftraining.com/
Which has a live training piece that will hopefully help fund some of my time to clean things up. If you're interested in contributing changes to the docs, or improving things in general, always happy to meet. It's much easier for me to donate time to review pull requests and provide direction than it is to write the code myself these days.

**Comment by kuehlc** [comment]
@dannyrb Fair enough. I respect that. Once I'm done with this big project (hopefully within a month or two), I'm happy to try to help with contributions in my free time.
